slatedb/src/merge_iterator.rs

```diff
@@ -157,22 +160,26 @@ impl KeyValueIterator for MergeIterator<'_> {
             return self.advance().await;
         }
 
-        let mut current_kv = match self.advance().await? {
+        let current_kv = match self.advance().await? {
             Some(kv) => kv,
             None => return Ok(None),
         };
 
-        // iterate until we find a key that is not the same as the current key,
-        // find the one with the highest seqnum.
-        while let Some(peeked_entry) = self.peek() {
-            if peeked_entry.key != current_kv.key {
-                break;
-            }
-            if peeked_entry.seq > current_kv.seq {
```

Comment:
I guess you're deleting this because it's guaranteed that peeked entry seq will be <  current kv when `peeked_entry.key == current_kv.key`?